openedx/features/subscriptions/utils.py

```diff
@@ -0,0 +1,31 @@
+from courseware.access_utils import ACCESS_DENIED, ACCESS_GRANTED
+from openedx.features.subscriptions.models import UserSubscription
+
+from student.models import User
+from enrollment.data import get_course_enrollment
+
+
+def track_subscription_enrollment(subscription_id, enrollment):
+    """
+    Add user enrollment to valid subscription.
+    """
+    if subscription_id:
+        username = enrollment.get('user')
+        user = User.objects.get(username=username)
+        valid_user_subscription = UserSubscription.get_valid_subscription(user.id)
+        if valid_user_subscription and valid_user_subscription.subscription_id == subscription_id:
```

Comment:
`valid_user_subscription` is a queryset so how we can directly use `valid_user_subscription.subscription_id`? shouldn't it be something like `valid_user_subscription.first().subscription_id`?

Reply:
I had modified `get_valid_subscription` method to return just one subscription when I wrote this function. the logic changed afterwards. I have fixed this.